src/Octoshift/Services/BbsApi.cs

```diff
@@ -73,11 +76,76 @@ public virtual async Task<IEnumerable<(int Id, string Key, string Name)>> GetPro
             .ToListAsync();
     }
 
+    public virtual async Task<(int Id, string Key, string Name)> GetProject(string projectKey)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}";
+        var response = await _client.GetAsync(url);
+
+        var project = JObject.Parse(response);
+        return ((int)project["id"], (string)project["key"], (string)project["name"]);
+    }
+
     public virtual async Task<IEnumerable<(int Id, string Slug, string Name)>> GetRepos(string projectKey)
     {
         var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos";
         return await _client.GetAllAsync(url)
             .Select(x => ((int)x["id"], (string)x["slug"], (string)x["name"]))
             .ToListAsync();
     }
+
+    public virtual async Task<bool> GetIsRepositoryArchived(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}?fields=archived";
+        var response = await _client.GetAsync(url);
+
+        var data = JObject.Parse(response);
+        return (bool)data["archived"];
+    }
+
+    public virtual async Task<IEnumerable<(int Id, string Name)>> GetRepositoryPullRequests(string projectKey, string repo)
+    {
+        var url = $"{_bbsBaseUrl}/rest/api/1.0/projects/{projectKey.EscapeDataString()}/repos/{repo.EscapeDataString()}/pull-requests?state=all";
```

Comment:
This is scope creep for this PR, so we don't need to address this now, but it would be great to use the `Uri` class for construction of these URLs at some point: https://learn.microsoft.com/en-us/dotnet/api/system.uri?view=net-7.0

I created an issue to address this later:

- https://github.com/github/gh-gei/issues/1040